File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class OrderStatusCount(BaseModel):
    status: str
    count: int


class DailyRevenue(BaseModel):
    date: str
    revenue: float
    order_count: int


class BestSellingFood(BaseModel):
    name: str
    quantity_sold: int
    revenue: float


class AnalyticsSummaryResponse(BaseModel):
    total_orders: int
    total_revenue: float
    total_customers: int
    total_delivery_partners: int
    orders_by_status: list[OrderStatusCount]
    revenue_last_7_days: list[DailyRevenue]
    best_selling_foods: list[BestSellingFood]

File: frontend/src/types/analytics.ts
export interface OrderStatusCount {
  status: string;
  count: number;
}

export interface DailyRevenue {
  date: string;
  revenue: number;
  order_count: number;
}

export interface BestSellingFood {
  name: string;
  quantity_sold: number;
  revenue: number;
}

export interface AnalyticsSummary {
  total_orders: number;
  total_revenue: number;
  total_customers: number;
  total_delivery_partners: number;
  orders_by_status: OrderStatusCount[];
  revenue_last_7_days: DailyRevenue[];
  best_selling_foods: BestSellingFood[];
}
